libp2p/transport/websocket/transport.py

```diff
@@ -1,229 +1,922 @@
+                        logger.debug("Using custom TLS client config")
+                    else:
+                        # Create default SSL context for client
+                        ssl_context = ssl.create_default_context()
                         ssl_context.check_hostname = False
                         ssl_context.verify_mode = ssl.CERT_NONE
+                        logger.debug("Using default TLS client config (insecure)")
+
+            # Handle proxy connections
+            if final_proxy_url:
+                logger.info(f"Using SOCKS proxy: {final_proxy_url} for {ws_url}")
+                self._proxy_connections += 1
+                conn = await self._create_proxy_connection(
+                    proto_info, final_proxy_url, ssl_context
+                )
+            else:
+                # Direct connection (no proxy)
+                logger.debug(f"Direct connection to {ws_url} (no proxy)")
+                conn = await self._create_direct_connection(proto_info, ssl_context)
 
-            logger.debug(f"WebsocketTransport.dial opening connection to {ws_url}")
+            if not conn:
+                raise OpenConnectionError(f"Failed to create connection to {ws_url}")
 
-            # Use a different approach: start background nursery that will persist
-            logger.debug("WebsocketTransport.dial establishing connection")
+            # Track connection
+            await self._track_connection(conn)
 
-            # Import trio-websocket functions
-            from trio_websocket import connect_websocket
-            from trio_websocket._impl import _url_to_host
+            logger.info(f"Connection established to {ws_url}")
+            return conn
 
-            # Parse the WebSocket URL to get host, port, resource
-            # like trio-websocket does
-            ws_host, ws_port, ws_resource, ws_ssl_context = _url_to_host(
-                ws_url, ssl_context
-            )
+        except trio.TooSlowError as e:
+            self._failed_connections += 1
+            logger.error(f"Connection timeout after {self._config.handshake_timeout}s")
+            raise OpenConnectionError(
+                f"WebSocket handshake timeout after {self._config.handshake_timeout}s"
+            ) from e
+        except Exception as e:
+            self._failed_connections += 1
+            logger.error(f"Failed to connect to {ws_url}: {e}", exc_info=True)
+            raise OpenConnectionError(f"Failed to connect to {ws_url}: {str(e)}")
+
+    async def _create_direct_connection(
+        self, proto_info: ParsedWebSocketMultiaddr, ssl_context: ssl.SSLContext | None
+    ) -> P2PWebSocketConnection:
+        """Create a direct WebSocket connection."""
+        # Extract host and port from the rest_multiaddr
+        host = (
+            proto_info.rest_multiaddr.value_for_protocol("ip4")
+            or proto_info.rest_multiaddr.value_for_protocol("ip6")
+            or "localhost"
```

Comment:
You include DNS in `_create_connection` above and `_create_proxy_connection` below, but not here. Is that intentional? (just curious, I really don't know)